### PR TITLE
Add more detailed documentation for fork and observe. Issue #561.

### PR DESCRIPTION
**Pull request by vqvu**
@gregory I took a stab at expanding the documentation for `fork` and `observer`.

What do you think?

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -1672,7 +1672,9 @@ Stream.prototype.write = function (x) {
 /**
  * Forks a stream, allowing you to add additional consumers with shared
  * back-pressure. A stream forked to multiple consumers will only pull values
- * from its source as fast as the slowest consumer can handle them.
+ * from its source as fast as the slowest consumer can handle them. For an
+ * example of the difference between `fork` and `observe`, see [this
+ * post](https://github.com/caolan/highland/issues/561#issuecomment-257069894).
```
Comment (gregory) [inline lib/index.js]: Is this needed if you include an example below?

File: lib/index.js [hunk 2]
```diff
@@ -1672,7 +1672,9 @@ Stream.prototype.write = function (x) {
 /**
  * Forks a stream, allowing you to add additional consumers with shared
  * back-pressure. A stream forked to multiple consumers will only pull values
```
Comment (gregory) [inline lib/index.js]: what about: 
` A stream forked to multiple consumers will only pull values, ONE AT A TIME, from its source as fast as the slowest consumer can handle them. `

Comment (vqvu) [reply lib/index.js]: Added.

File: lib/index.js (hunk 1 above)
Comment (vqvu) [reply lib/index.js]: Good point. I've incorporated more of the example from that comment directly into the docs and then removed the link to the comment.

**Comment by gregory** [comment]
neat - thanks @vqvu !

**Pull request merged**